Drop glibc as a runtime dependency

mrjackv opened the pull request.

mrjackv committed:
Drop any runtime dependency on glibc

The glibc component should be used exclusively for linking to when
building components inside revng. Drop the instances where a component
declares requiring glibc at runtime

mrjackv committed:
linux_headers.lib.yml: Fix url

This commit removes the hardcoded major number that was present in the
source_url

aleclearmind commented on .orchestra/config/components/libedit.yml:
```diff
@@ -1,32 +0,0 @@
-#@overlay/match-child-defaults missing_ok=True
----
-components:
-  libedit: #@ single_build_component(**libedit_args())
```
Please drop this commit, it's an optional dependency of both `llvm` and `clang-release`. I noticed it recently in an old llvm build directory which complained about missing libedit since we no longer pull it from the host system.

In fact, can you add it to those components?

mrjackv replied:
Added it back and enabled it in both `clang-release` and `llvm`

aleclearmind commented:
Nice!

mrjackv committed:
libedit: add as a dependency of clang/llvm

The libedit component was unused since the dependency on it from
clang/llvm was not implemented.

mrjackv committed:
wine: force `WINE{PREFIX,ARCH}`

When running executables under wine, force the use of the wine prefix
`$ORCHESTRA_DOTDIR/wine-prefix` to ensure consistency and avoid
littering the user's default wine prefix

aleclearmind commented on .orchestra/config/components.yml:
```diff
@@ -17,6 +17,8 @@ environment:
   - ORCHESTRA_NODE_CACHE: $ORCHESTRA_ROOT/share/node_cache
   - HARD_FLAGS_YARN: "--non-interactive --no-progress --frozen-lockfile --offline"
   - TERMINFO: $ORCHESTRA_ROOT/share/terminfo
+  - WINEPREFIX: $ORCHESTRA_DOTDIR/wine-prefix
```
What do you think of putting this in `.orchestra/cache`? So users can safely assume it's safe to drop it.

mrjackv replied:
We could do that, but isn't `.orchestra/cache` for orchestra-the-tool to store its data only?
(Nothing against it in principle, just making sure there isn't some assumption that it should not be touched by other entities)

aleclearmind replied:
OK, let's not touch for now. But we need to concentrate all the disposable stuff in a single directory at a certain point (include `binary-archives`, `source-archives`, `tmproot`, `config_cache.json`, `config_cache.yml`, and `remote_refs_cache.json`).

But not now.

aleclearmind commented:
Merged in 4f1a7f8, thanks!
